File: v6/sprefa-extract/tests/fixtures/tsi/probe_graph.ts
// Every shape the ts syntax tier claims, in one file: a class with an
// application, a tuple, a literal and a function-typed field; an interface
// with an optional and a readonly member; a union alias and an applied alias;
// a typed const and a typed let; a function with keyword types.

export class Error {}

export class Trail<T> {
  steps: Map<string, Option<T>>;
  label: [string, number];
  failed: { reason: string; code?: number };
  project: (element: T) => U;
  outcome: Result<number, Error>;
}

export interface Render {
  readonly id: string;
  name?: string;
  render(into: Formatter): boolean;
}

export type Step = "idle" | "retry" | "failed";

export type Query = Partial<Trail<number>>;

export const RETRY_LIMIT: number = 3;

export let banner: string;

export function isEmpty(steps: string[], flag: boolean): void {}

File: v6/sprefa-extract/tests/fixtures/tsi/probe_graph.rs
// Every shape the rust syntax tier claims, in one file: the three variant
// forms, a generic product, a const, a static, an inherent impl, a trait.

use std::fmt;

pub struct Error;

pub enum Step {
    Idle,
    Retry(u32, bool),
    Failed { reason: String, code: u32 },
}

pub struct Trail<T> {
    pub steps: Vec<Option<T>>,
    pub outcome: Result<u64, Error>,
    pub label: (String, u32),
    pub tag: Box<str>,
    pub rendered: std::fmt::Result,
}

pub const RETRY_LIMIT: u32 = 3;

pub static BANNER: &str = "trail";

impl<T> Trail<T> {
    pub fn is_empty(&self) -> bool {
        self.steps.is_empty()
    }

    pub fn clear(&mut self) -> () {}
}

pub trait Render {
    fn render(&self, into: &mut fmt::Formatter) -> bool;
}

impl<T> Render for Trail<T> {
    fn render(&self, into: &mut fmt::Formatter) -> bool {
        true
    }
}
